src/socket.ts

```diff
@@ -36,23 +36,26 @@ export function validateCertificate(
   dnsName: string
 ) {
   return (hostname: string, cert: tls.PeerCertificate): Error | undefined => {
-    if (serverCaMode === 'GOOGLE_MANAGED_CAS_CA') {
+    if (!serverCaMode || serverCaMode !== 'GOOGLE_MANAGED_INTERNAL_CA') {
```

Comment:
i wonder if for Node you be fine removing the `!serverCaMode` and having purely
```suggestion
    if (serverCaMode !== 'GOOGLE_MANAGED_INTERNAL_CA') {
```

Since i believe it will be `undefined` which should also be captured by the ` !==`

Reply:
Tests were failing due to a logic error. I fixed it: `!serverCaMode || serverCaMode === 'GOOGLE_MANAGED_INTERNAL_CA'`